test/e2e/storage/vsphere/connection.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"fmt"
+	neturl "net/url"
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/session"
+	"github.com/vmware/govmomi/vim25"
+	"golang.org/x/net/context"
+)
+
+const (
+	roundTripperDefaultCount = 3
+)
+
+var (
+	clientLock sync.Mutex
+)
+
+// Connect makes connection to vSphere
+// No actions are taken if a connection exists and alive. Otherwise, a new client will be created.
+func Connect(ctx context.Context, vs *VSphere) error {
+	var err error
+	clientLock.Lock()
+	defer clientLock.Unlock()
+
+	if vs.Client == nil {
+		vs.Client, err = NewClient(ctx, vs)
+		if err != nil {
+			glog.Errorf("Failed to create govmomi client. err: %+v", err)
+			return err
+		}
+		return nil
+	}
+	m := session.NewManager(vs.Client.Client)
```

Comment:
nit: can rename var to manager. m is too short.

Reply:
Done.